frontend/src/services/socket.ts

```diff
@@ -41,7 +41,7 @@ class Socket {
   private static _initialize(token: string): void {
     if (Socket.isConnected()) return;
 
-    const WS_URL = `ws://${window.location.host}/ws?token=${token}`;
+    const WS_URL = `${window.location.protocol}//${window.location.host}/ws?token=${token}`;
```

Comment:
Shouldn't it be something like
```bash
const protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
const WS_URL = `${protocol}//${window.location.host}/ws?token=${token}`;
```

Reply:
Yes

Reply:
https://haacked.com/archive/2019/06/03/suggested-changes/